faculty/clients/environment.py

```diff
@@ -42,16 +186,77 @@ class EnvironmentSchema(BaseSchema):
     author_id = fields.UUID(data_key="authorId", required=True)
     created_at = fields.DateTime(data_key="createdAt", required=True)
     updated_at = fields.DateTime(data_key="updatedAt", required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
+    def make_environment(self, data):
+        return Environment(**data)
+
+
+class EnvironmentUpdateSchema(BaseSchema):
+    name = fields.String(required=True)
+    description = fields.String(required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
+    def make_environment_update(self, data):
+        return EnvironmentUpdate(**data)
+
+
+class EnvironmentCreationResponseSchema(BaseSchema):
+    id = fields.UUID(data_key="environmentId", required=True)
+    project_id = fields.UUID(data_key="projectId", required=True)
+    name = fields.String(required=True)
+    description = fields.String(required=True)
+    author_id = fields.UUID(data_key="authorId", required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
 
     @post_load
     def make_environment(self, data):
+        data["created_at"] = None
+        data["updated_at"] = None
         return Environment(**data)
 
 
+class DummySchema(BaseSchema):
+    def load(self, in_data):
+        return None
+
+
 class EnvironmentClient(BaseClient):
 
     SERVICE_NAME = "baskerville"
 
     def list(self, project_id):
         endpoint = "/project/{}/environment".format(project_id)
         return self._get(endpoint, EnvironmentSchema(many=True))
+
+    def get(self, project_id, environment_id):
+        endpoint = "/project/{}/environment/{}".format(
+            project_id, environment_id
+        )
+        return self._get(endpoint, EnvironmentSchema())
+
+    def update(self, project_id, environment_id, update):
+        endpoint = "/project/{}/environment/{}".format(
+            project_id, environment_id
+        )
+        return self._put(
```

Comment:
If we don't care about the return, I think you can just use `self._put_raw`. It then doesn't try to do anything with the return value.